app-emulation/virtualbox-guest-additions/virtualbox-guest-additions-7.0.8-r2.ebuild

```diff
@@ -0,0 +1,304 @@
+# Copyright 2022-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit edo linux-mod-r1 systemd toolchain-funcs udev
+
+MY_PN="VirtualBox"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="VirtualBox kernel modules and user-space tools for Gentoo guests"
+HOMEPAGE="https://www.virtualbox.org/"
+SRC_URI="https://download.virtualbox.org/virtualbox/${PV}/${MY_P}a.tar.bz2
+	https://gitweb.gentoo.org/proj/virtualbox-patches.git/snapshot/virtualbox-patches-7.0.8.tar.bz2"
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+# Reminder: see the LICENSE related comment in app-emulation/virtualbox-additions ebuild
+LICENSE="GPL-3 LGPL-2.1+ MIT || ( GPL-3 CDDL )"
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="~amd64 ~x86"
+IUSE="+dbus gui"
+
+# automount Error: VBoxServiceAutoMountWorker: Group "vboxsf" does not exist
+# TODO: find out what this is, remove comment if obsolete
+
+RDEPEND="
+	acct-group/vboxguest
+	acct-group/vboxsf
+	acct-user/vboxguest
+	sys-libs/pam
+	sys-libs/zlib
+	dbus? ( sys-apps/dbus )
+	gui? (
+		x11-apps/xrandr
+		x11-apps/xrefresh
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXmu
+		x11-libs/libXt
+	)
+"
+# some libs here are indirect dependencies, and also needed at compile time.
+# keeping them in DEPEND to avoid warnings from qa-vdb.
+DEPEND="
+	${RDEPEND}
+	gui? (
+		x11-libs/libICE
+		x11-libs/libSM
+		x11-libs/libXau
+		x11-libs/libXdmcp
+		x11-base/xorg-proto
+	)
+"
+BDEPEND="
+	>=dev-lang/yasm-0.6.2
+	>=dev-util/kbuild-0.1.9998.3127
+	sys-devel/bin86
+	sys-power/iasl
+"
+PDEPEND="
+	gui? ( x11-drivers/xf86-video-vboxvideo )
+"
+
+PATCHES=(
+	"${FILESDIR}"/virtualbox-guest-additions-7.0.8-kernel-6.4-patch-1.patch
+	"${FILESDIR}"/virtualbox-guest-additions-7.0.8-kernel-6.4-patch-2.patch
+)
+
+VBOX_MOD_SRC_DIR="${S}/out/linux.${ARCH}/release/bin/additions/src"
```

Comment:
Doesn't really matter given you set S yourself, but fwiw referencing `${S}` is not legal in global scope if we go by PMS specs (portage does always have it populated, but other tools or package managers could do whatever they want).

https://projects.gentoo.org/pms/8/pms.html#x1-109001r1

It's a common nitpick with ebuilds that used linux-mod-r0.

I'd either move set it inside a phase (it will be readable by other phases regardless if not local), and/or not use ${S} given all phases are already relative to `${S}`.

Alternatively inlining the path may not be so bad, it's not repeated that much (see later comment wrt its use in src_install).

Reply:
I removed the `${S}`, it's relative to it anyway